service/src/main/java/uk/nhs/adaptors/gp2gp/gpc/GpcClient.java

```diff
@@ -20,8 +21,19 @@ public String getStructuredRecord(RequestHeadersSpec<? extends RequestHeadersSpe
             structuredTaskDefinition.getFromAsid(),
             gpcConfiguration.getUrl() + gpcConfiguration.getStructuredEndpoint());
 
-        return request
-            .retrieve()
+        return request.retrieve()
+            .bodyToMono(String.class)
+            .block();
+    }
+
+    public String getDocumentRecord(WebClient.RequestHeadersSpec<? extends WebClient.RequestHeadersSpec<?>> request,
+            GetGpcDocumentTaskDefinition documentTaskDefinition) {
```

Comment:
Could change this two TaskDefinition variable in prams and reuse the same method, then env variable for type of request maybe for logger?

Reply:
Ok